Jetstream/Client.swift

```diff
@@ -72,16 +75,22 @@ public enum ClientStatus {
     }
     
     let logger = Logging.loggerFor("Client")
-    let transport: Transport
+    let transportAdapterFactory: TransportAdapterFactory
+    let restartSessionOnFatalError: Bool
+    var transport: Transport
     var sessionCreateParams = [String: AnyObject]()
+    var explicitlyClosed = false
     
     // MARK: - Public interface
     
     /// Constructs a client.
     ///
     /// :param: transportAdapter The transport adapter to use to connect to the Jetstream server.
-    public init(transportAdapter: TransportAdapter) {
-        transport = Transport(adapter: transportAdapter)
+    /// :param: restartSessionOnFatalError Whether to restart a session on a fatal session or transport error.
+    public init(transportAdapterFactory: TransportAdapterFactory, restartSessionOnFatalError: Bool = true) {
```

Comment:
Should the transportAdapterFactory be the last argument as is usually the case with closures?